src/main/kotlin/it/pagopa/wallet/repositories/PaymentMethodsTemplateWrapper.kt

```diff
@@ -0,0 +1,21 @@
+package it.pagopa.wallet.repositories
+
+import it.pagopa.generated.ecommerce.model.PaymentMethodResponse
+import java.time.Duration
+import org.springframework.data.redis.core.RedisTemplate
+
+class PaymentMethodsTemplateWrapper
+/**
+ * Primary constructor
+ *
+ * @param redisTemplate inner redis template
+ * @param ttl time to live for keys
+ */
+(redisTemplate: RedisTemplate<String, PaymentMethodResponse>, ttl: Duration) :
+    RedisTemplateWrapper<PaymentMethodResponse>(
+        redisTemplate = redisTemplate,
+        "payment-methods",
```

Comment:
Could it be a good idea saving the cache per client? Something like "wallet-service:payment-methods"

Reply:
This cache will be only for wallet-service. Which other kind of client do you guess to use this cache ?

Reply:
I assume that payment-methods cache can be used by other services too. Do you agree with this?

Reply:
We are saving an entry on the Redis cache that is generic ("payment-methods") and it is handled by a service (wallet-service) in both reads and writes; even if there is only one client that uses the payment-methods response at the moment, we could already save it by client in order to avoid accessing the same entry from multiple applications in the future (if there will be any new client that has to access to the same data). I'm just suggesting a cache by client instead of a cache for all.

Reply:
Done here 4211527ba8a8bba7f7f74740c1d3a4febb22609c